ethstorage/miner/l1_mining_api.go

```diff
@@ -112,12 +115,59 @@ func (m *l1MiningAPI) SubmitMinedResult(ctx context.Context, contract common.Add
 	}
 	m.lg.Info("Estimated gas done", "gas", estimatedGas)
 
+	nonce, err := m.NonceAt(ctx, cfg.SignerAddr, big.NewInt(rpc.LatestBlockNumber.Int64()))
+	if err != nil {
+		m.lg.Error("Query nonce failed", "error", err.Error())
+		return common.Hash{}, err
+	}
+	m.lg.Debug("Query nonce done", "nonce", nonce)
+	safeGas := uint64(float64(estimatedGas) * gasBufferRatio)
+
+	// Check gas fee against tx fee cap (e.g., --rpc.txfeecap as in geth) early to avoid tx fail)
+	txFeeCapDefault := ether
+	if m.rc != nil {
+		txFeeCapDefault = txFeeCapDefaultL2
+	}
+	txFee := new(big.Int).Mul(new(big.Int).SetUint64(safeGas), gasFeeCap)
+	m.lg.Debug("Estimated tx fee on the safe side", "safeGas", safeGas, "txFee", txFee)
+	if txFee.Cmp(txFeeCapDefault) == 1 {
+		m.lg.Error("Mining tx dropped: tx fee exceeds the configured cap", "txFee", fmtEth(txFee), "cap", fmtEth(txFeeCapDefault))
+		return common.Hash{}, errDropped
+	}
+
+	rawTx := &types.DynamicFeeTx{
+		ChainID:   m.NetworkID,
+		Nonce:     nonce,
+		GasTipCap: tip,
+		GasFeeCap: gasFeeCap,
+		Gas:       safeGas,
+		To:        &contract,
+		Value:     common.Big0,
+		Data:      calldata,
+	}
+
+	unsignedTx := types.NewTx(rawTx)
+	extraCost := new(big.Int)
+	// Add L1 data fee as tx cost when es-node is deployed as an L3
+	if m.rc != nil {
+		l1fee, err := m.getL1Fee(ctx, unsignedTx)
```

Comment:
Is there any difference between unsignedTx and signedTx? If so, I believe using signedTx would be more accurate

Reply:
Using signed tx may return a different fee from unsigned, but it looks like the function requires unsigned tx: https://github.com/ethereum-optimism/optimism/blob/39e9f1a4693912af8bdef428d246c15a6cf44ec7/packages/contracts-bedrock/src/L2/GasPriceOracle.sol#L55